### DISCUSSION TITLE
How is the intensity in decibel computed?

### DISCUSSION THREAD

**Discussion by vijaygirish2001** [YannickJadoul/Parselmouth#62] [Q&A] [answered]
I use the Parselmouth's intensity = snd.to_intensity() function for computing the intensity. Can anyone let me know the formula/ expression for the intensity calculation function, and the frame size used?

I see intensity in decibel is computed as:

10 log(Mean energy/ **Constant**) 

where Mean energy is frame level energy, I assume default frame size is 40 ms

If my assumption about the above expression is correct, what is the value of **Constant** used ?

**Comment by YannickJadoul** [comment, marked as answer]
Hi, @vijaygirish2001. Yes, the threshold is 4.0e-10 (or 2.0e-5 squared). The relevant part of Praat's code is here:

https://github.com/YannickJadoul/Parselmouth/blob/6111e6e1c7dd75b9a06df0e81b2c0bd45776d13d/praat/fon/Sound_to_Intensity.cpp#L115-L120

The Praat manual also very shortly describes it here: https://www.fon.hum.uva.nl/praat/manual/Intensity.html. And details about the window size and time step are here: https://www.fon.hum.uva.nl/praat/manual/Sound__To_Intensity___.html.
https://www.fon.hum.uva.nl/praat/manual/sound_pressure_calibration.html discusses how these values are to some degree arbitrary for recorded audio, though, and how Praat interprets the amplitude values of the audio file as pressure values.
